### ISSUE TITLE
Expose font metrics in a standardized way

### ISSUE THREAD

**Issue by AaronDavidNewman** [0xfe/vexflow#810]
I propose creating a simple TextFont object that acts as a registry for the default fonts, as well as any other fonts an application may want to use.  The application would register their preferred font metrics in the same way the default ones are registered, and include the font data any way they like (webfont, static, etc.).

Benefits might be:
* use font metrics in other text modules like annotations.  Currently these modules can't calculate the metrics pre-render and have to guess.
* automate basic calculations like width in pixels of a glyph.  If exact metrics are not available for a glyph, use a fallback glyph
* Vex applications can use non-English fonts
* vex applications can create font choosers with available fonts

Registering a font would be like:
``` javaascript
     TextFont.registerFont({
        name: 'RobotoSlabMetrics',
        resolution: RobotoSlabMetrics.resolution,
        glyphs: RobotoSlabMetrics.glyphs,
        fontFamily: RobotoSlabMetrics.fontFamily,
        serifs: true,
        monospaced: false,
        maxSizeGlyph: 'H',
        superscriptOffset: 0.66,
        subscriptOffset: 0.66,
        description: 'Default serif text font to pair with Bravura/Gonville engraving font',
      });
```
Fields like `serifs` and `descriptions` are informational and can be used by applications for font-list dialogs, or permit selecting a fallback font dynamically.

A font instance would be the font metrics combined with a particular font size (which could be changed dynamically).

``` javascript
const myFont = new TextFont({name: 'RobotoSlab', size: 12});
// ...
width += myFont.getWidthForGlyph('w');
```

I considered adding kerning logic here also.   I think that would be good, but what I do in Chord symbols isn't _really_ kerning.  Kerning and ligatures would be another effort.

**Comment by 0xfe** [comment]
@AaronDavidNewman This sound like a great idea! We might also want a `TextGlyph` which is a special case of `Glyph` with text specific methods.

**Comment by AaronDavidNewman** [comment]
>> We might also want a `TextGlyph` which is a special case of `Glyph`

I had considered this, but I couldn't think of a use case for it.  Like, I can't think of the case where 'rendering' a text glyph is anything other than putting the character in a 'text' element.  Is this just to be compatible with the existing `Glyph` interface?     I could see some use in that, but you'd still need to distinguish at some level what type of glyph it was before rendering, or else include all the path data along with the metrics.

I created a version of the fontgen tool that generates the metrics that I thought would be useful for text fonts, and I removed the path generation because I didn't think it would be useful.
